api/src/main/java/com/datatorrent/api/StringCodec.java

```diff
@@ -377,4 +384,43 @@ public String toString(Class<? extends T> clazz)
     private static final long serialVersionUID = 201312082053L;
   }
 
+  public class JsonStringCodec<T> implements StringCodec<T>, Serializable
+  {
+    private static final long serialVersionUID = 2513932518264776006L;
+    Class<?> clazz;
+
+    public JsonStringCodec(Class<T> clazz)
+    {
+      this.clazz = clazz;
+    }
+
+    @Override
+    public T fromString(String string)
+    {
+      try {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.configure(DeserializationConfig.Feature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        ObjectReader reader = mapper.reader(clazz);
+        return reader.readValue(string);
+      } catch (IOException e) {
+        Throwables.propagate(e);
```

Comment:
Would be nice to fix the other occurrences in this file also and remove usage of DTThrowable.